u2f-ref-code/java/src/com/google/u2f/server/messages/RegisteredKey.java

```diff
@@ -111,9 +111,11 @@ private JsonArray getTransportsAsJson() {
     return transportsArray;
   }
 
-  public JsonObject getJson() {
+  public JsonObject getJson(String defaultAppId) {
     JsonObject result = new JsonObject();
-    result.addProperty("appId", appId);
+    if (appId != null && !appId.equals(defaultAppId)) {
```

Comment:
Is that necessary? The specs don't allow you to set an empty value, and the client should reject it.